bigquery/storage/managedwriter/managed_stream_test.go

```diff
@@ -539,12 +541,24 @@ func TestManagedStream_LeakingGoroutinesReconnect(t *testing.T) {
 		if testArc.openCount != i+2 {
 			t.Errorf("should trigger a reconnect, but found openCount %d", testArc.openCount)
 		}
+		// Bump our threshold each append.  We add 2 goroutines each reconnect due to a new processor
+		// and a graceful cancellation, but this should recover over time.
```

Comment:
"should" doing a lot of heavy lifting here 😁  Are there conditions in which this would not recover over time?